### ISSUE TITLE
Rethink meaning of "latest/previous value"

### ISSUE THREAD

**Issue by pozadi** [pozadi/kefir#102]
We have number of places with some behavior defined by "latest/previous value" of some observable. 
For instance in `.combine` you get **latest values** of all sources in combinator function. 
Or in `.scan` you get **previous value** in handler function.

Also besides `value-events` we have `error-events`. But now it works like a two parallel streams of events, so when we consider **latest value** somewhere we don't take into account errors: if the latest `event` was a `error-event`, we still use **value** from previous `value-event` as **latest value**.

So now it looks like this:
```
values         -----1----2--------3---
errors         -------------e---------   // basically ignored
latest value   _____111112222222223333
```

But should be:
```
values/errors  -----1----2--e-----3---
latest value   _____11111222______3333
```
(`_` means that value is undefined)

Basically we should store not **latest value**, but **latest event**, which can be value or error.

This was already fixed in [Properties](https://github.com/pozadi/kefir/issues/55) and [.combine](https://github.com/pozadi/kefir/issues/54) in version 2.0.0, but we have other methods that also operates on **latest/previous value**, for instance `.scan`.

Also the latest value should became undefined on each deactivation, this [was planned to be fixed in property](https://github.com/pozadi/kefir/issues/59), but should be fixed in all places where we use **latest/previous value** somehow.

**Comment by pozadi** [comment]
Any thoughts on why it might be a bad idea?

**Comment by rpominov** [comment]
Another solution we should consider is to simply end on a first error, like RxJS, Most, etc. do. This way problem automatically goes away.

**Comment by rpominov** [comment]
`.scan` already fallbacks to the seed after an error (it got this behavior after fixing #148)
